### ISSUE TITLE
display the 'doc' field from the .ksy file in the JSON

### ISSUE THREAD

**Issue by shengbinxu** [kaitai-io/kaitai_struct#1077]
My requirements: Convert the .ksy file into a JSON file that is more readable.

Is it possible to display the 'doc' field from the .ksy file in the JSON?
If not, are there any other methods? For example, defining a field under 'instances' and setting its value to a fixed string.

```ksy
    instances:
      doc:
        value: 'the document of field'
```

**Comment by generalmimon** [comment]
> My requirements: Convert the .ksy file into a JSON file that is more readable.

I don't think it will necessarily be more readable than the .ksy file (but I guess that's up to you), but it's quite trivial to do that. KSY stands for Kaitai Struct YAML, meaning that .ksy files are indeed based on [YAML](https://learnxinyminutes.com/docs/yaml/). So if you want to convert .ksy files to JSON, you just need a converter from YAML to JSON (you can also do this from any programming language that has a YAML parser and JSON serializer).

> Is it possible to display the 'doc' field from the .ksy file in the JSON?

To be honest, I don't know what kind of JSON you're talking about. This last sentence of yours:

> If not, are there any other methods? For example, defining a field under 'instances' and setting its value to a fixed string.
> 
> ```
>     instances:
>       doc:
>         value: 'the document of field'
> ```

... makes me think that you mean the JSON dump of the object tree with parsed data from a binary file, as provided by the "export to JSON" feature of the [Web IDE](https://ide.kaitai.io/devel/) or the [`ksdump`](https://github.com/kaitai-io/kaitai_struct_visualizer?tab=readme-ov-file#usage) command-line tool.

If so, yes, you can use the approach you've suggested, but you need one extra pair of quotes (the outer pair is because of [YAML](https://learnxinyminutes.com/docs/yaml/), the inner pair is for the KS expression language):

```ksy
    instances:
      doc:
        value: '"the document of field"'
```

**Comment by generalmimon** [comment]
I consider this question answered and there has been no activity for over a year, so I am closing this issue.
